Material EditText

rastislavvasko opened the pull request:
This adds a `MaterialEditText` with a tinted background and cursor.

Also added is partial support for the `android:theme` attribute for widgets. Partial, because the themed context won't be passed to the widget's descendents.

Lastly, `MaterialWidgetHandler` now supports initializing drawables by raw color value.

rastislavvasko committed:
Updates Support and AppCompat libs to rev 22.

rastislavvasko committed:
Adds InsetMaterialDrawable.

rastislavvasko committed:
Adds MaterialEditText with a tinted background.

rastislavvasko committed:
Adds RobotoEditText to sample activity.

rastislavvasko committed:
Backports tinted cursor to MaterialEditText.

rastislavvasko committed:
Code style update.

rastislavvasko committed:
Adds partial support for android:theme.

The android:theme support is only partial, because the themed context won't be passed to the widget's descendents.

Also added is support for initializing drawables by raw color value.

rastislavvasko committed:
Themified MaterialEditText.

jpmcosta commented on MaterialLibrary/src/main/java/io/doist/material/widget/RobotoEditText.java:
```diff
@@ -3,11 +3,10 @@
 import io.doist.material.utils.Roboto;
 
-public class RobotoEditText extends EditText {
+public class RobotoEditText extends MaterialEditText {
```
I think that `MaterialEditText` is the one that should extend from `RobotoEditText`.

jpmcosta commented on MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java:
```diff
@@ -1,27 +1,34 @@
 
 public class MaterialWidgetHandler {
-    private static final boolean sSkip = Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT;
+    private static final boolean SKIP = Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT;
```
:+1:

jpmcosta commented on MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java:
```diff
@@ -151,4 +195,40 @@ public static void init(View view, AttributeSet set, int defStyle, int[] attrs)
         return newArray;
     }
+
+    private static Drawable getDrawable(MaterialResources r, TypedArray ta, int index) {
```
I don't think `MaterialWidgetHandler` should handle any part of `getDrawable`. We should always use `MaterialResources`. More so after the merge with https://github.com/Doist/AndroidMaterial/commit/eafcefa0694ea20aafb9ababfcf9bf075c113bd3.
What do you think?

jpmcosta commented:
This seems good.

Some notes:
- I believe that we won't be needing `MaterialContextThemeWrapper` after the merge with `ft.material_dialog`, since we will use reflection to get the theme from a `ContextThemeWrapper`;
- I think that `ft.material_dialog` version of `MaterialResources` can safely replace this branch's version. However, it would be very nice if you could review it, to make sure it is working and whether or not it covers all the functionality needed here.

We should obviously discuss any matter you find relevant, because my comments are based on my idea on how we can merge this branch with `ft.material_dialog`.

rastislavvasko committed:
Merge branch 'master' into ft.material_edit_text

Conflicts:
	MaterialLibrary/src/main/java/io/doist/material/drawable/InsetMaterialDrawable.java
	MaterialLibrary/src/main/java/io/doist/material/drawable/MaterialDrawableUtils.java
	MaterialLibrary/src/main/java/io/doist/material/res/MaterialResources.java
	MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java
	MaterialLibrary/src/main/res/values/attrs_material.xml
	MaterialLibrary/src/main/res/values/themes_material.xml

rastislavvasko committed:
Swaps MaterialEditText and RobotoEditText in class hierarchy.

rastislavvasko committed:
Restores tinted text cursor support.

rastislavvasko committed:
Restores partial android:theme support for MaterialEditText.

rastislavvasko committed:
Apply android:color attribute when inflating RippleMaterialDrawable.

rastislavvasko committed:
Adds android:theme support to MaterialTextView.

rastislavvasko committed:
Adds MaterialCheckBox widget.

rastislavvasko committed:
Adds MaterialRadioButton widget.

rastislavvasko commented:
This is now based on `ft.material_dialog` (after it was merged into `master`).

In fd9ff8da684824f1d6117362cdee65a031a4f276 the previous changes to `MaterialResources` and `MaterialWidgetHandler` were completely replaced with the version in `master` and later commits only restore the necessary functionality to make reviewing easier.

jpmcosta commented on MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java:
```diff
@@ -256,4 +350,21 @@ private static void initTextViewAttributes(Context context, MaterialResources re
         return newStyleable;
     }
+
+    private static int getThemeResId(Context context) {
```
We can probably turn `MaterialResources$getThemeResId(Context)` into a public method. What do you think?

jpmcosta commented:
The PR seems good.
Some notes:
- I think that we should add `themifyContext` to all material widgets, like `MaterialButton`, `MaterialEditText` and `MaterialView`.
- I also believe we should standardize our calls to `super()` or `this()` in the `Material<Widget>` constructors. I think we should use `this()` whenever possible.
- I think we could also fit a small change to `RobotoEditText` and `RobotoTextView` – when calling `applyRoboto` the default `defStyle` should be `android.R.attr.editTextStyle` and `android.R.attr.textViewStyle`, respectively. Currently we are just passing `0`.

rastislavvasko committed:
Removes duplicate getThemeResId method.

rastislavvasko committed:
Standardizes constructors of material widgets.

rastislavvasko committed:
Standardizes constructors of roboto widgets.

rastislavvasko committed:
Adds support for android:theme to some material widgets.

rastislavvasko committed:
Adds support for android:theme to FloatingActionButton.

Also replaces the FAB's android:onClick with a listener, because android:theme makes android:onClick crash.

rastislavvasko commented:
Thanks for review! Fully agree with feedback and everything has been implemented. :)

jpmcosta commented:
Everything good! :+1:
